ui/components/src/LegendOptionsEditor/LegendOptionsEditor.tsx

```diff
@@ -0,0 +1,86 @@
+// Copyright 2022 The Perses Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import { Autocomplete, Switch, SwitchProps, TextField } from '@mui/material';
+import { DEFAULT_LEGEND_POSITION, LegendOptions } from '../model';
+import { OptionsEditorControl } from '../OptionsEditorLayout';
+
+type LegendPositionConfig = {
+  label: string;
+};
+
+const LEGEND_POSITIONS = ['bottom', 'right'] as const;
+
+type LegendPositions = typeof LEGEND_POSITIONS[number];
+
+const LEGEND_POSITIONS_CONFIG: Readonly<Record<LegendPositions, LegendPositionConfig>> = {
+  bottom: { label: 'Bottom' },
+  right: { label: 'Right' },
+};
+
+type LegendPositionOption = LegendPositionConfig & { id: LegendOptions['position'] };
+
+const POSITION_OPTIONS: LegendPositionOption[] = Object.entries(LEGEND_POSITIONS_CONFIG).map(([id, config]) => {
+  return {
+    id: id as LegendOptions['position'],
+    ...config,
+  };
+});
+
+export interface LegendOptionsEditorProps {
+  value?: LegendOptions;
+  onChange: (legend?: LegendOptions) => void;
+}
+
+export function LegendOptionsEditor({ value, onChange }: LegendOptionsEditorProps) {
+  const handleLegendShowChange: SwitchProps['onChange'] = (_: unknown, checked: boolean) => {
+    // legend is hidden when legend obj is undefined
+    const legendValue = checked === true ? {} : undefined;
```

Comment:
Nit: this is really about the overall `LegendOptions` type I think, but seems weird to have an empty object be a valid value here (I think I expected `position` to be required).

Reply:
Yeah this makes sense! It was pretty quick to change, I did it as a [separate PR here](https://github.com/perses/perses/pull/848) to be safe.